src/components/Checkbox.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { StyleSheet, Text, Pressable, View } from 'react-native'
+
+import { colors } from '../config/themes/appThemes'
+
+interface CheckboxProps {
+  onChange: (value: string) => void
+  value: string
+}
+
+export const Checkbox: React.FC<CheckboxProps> = ({ onChange, value }) => {
+  const [isChecked, setIsChecked] = React.useState(false)
```

Comment:
Solo importar el `useState` en vez de `React`. Ademas, siempre tipar 

```suggestion
  const [isChecked, setIsChecked] = useState<boolean>(false)
```